### PR TITLE
fix(http): do not allow encoded callback params in jsonp requests

### PR DESCRIPTION
**Pull request by petebacondarwin**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

fix

**What is the current behavior? (You can also link to an open issue here)**

security vector open in jsonp

**What is the new behavior (if this is a feature change)?**

vector is closed

**Does this PR introduce a breaking change?**

No

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

**Comment by Narretz** [comment]
LGTM

File: src/ng/http.js [hunk 1]
```diff
@@ -1420,20 +1420,21 @@ function $HttpProvider() {
...
-        throw $httpMinErr('badjsonp', 'Illegal use of JSON_CALLBACK in url, "{0}"', url);
-      }
-
-      var callbackParamRegex = new RegExp('[&?]' + key + '=');
-      if (callbackParamRegex.test(url)) {
-        // Throw if the callback param was already provided
-        throw $httpMinErr('badjsonp', 'Illegal use of callback param, "{0}", in url, "{1}"', key, url);
-      }
+    function sanitizeJsonpCallbackParam(url, cbKey) {
+      var params = parseKeyValue(url.split('?', 2)[1]);
```
Comment (gkalpak) [inline src/ng/http.js]: I think this will allow things like `...?foo=bar?&JSON_CALLBACK=...`.

Comment (petebacondarwin) [reply src/ng/http.js]: Hmm, so I guess we should error if there is more than one `?` in the URL?

Comment (gkalpak) [reply src/ng/http.js]: Works for me :+1:

**Comment by petebacondarwin** [comment]
Cherry picked to v1.6.x as 569e906a5818271416ad0b749be2f58dc34938bd

**Pull request merged**